src/Main.cc

```diff
@@ -470,7 +476,16 @@ void OnMessage(uWS::WebSocket<false, true>* ws, std::string_view sv_message, uWS
         }
     } else if (op_code == uWS::OpCode::TEXT) {
         if (sv_message == "PING") {
-            ws->send("PONG", uWS::OpCode::TEXT);
+            auto t_session = session->GetLastMessageTimestamp();
+            auto t_now = std::chrono::high_resolution_clock::now();
+            auto dt = std::chrono::duration_cast<std::chrono::seconds>(t_now - t_session);
+            if ((settings.idle_session_timeout >= 0) && (dt.count() >= settings.idle_session_timeout)) {
```

Comment:
I think we should use > 0. Using `0` will cause the session to be disconnected at every `PING` message